File: python/codingBat/String-2/xyz_there.py
#!/usr/bin/python3
# Return True if the given string contains an appearance of "xyz" where the xyz is not directly preceeded by a period (.). 
# So "xxyz" counts but "x.xyz" does not.

def xyz_there(str):
  if len(str) < 3: return False
  elif str[0:3].endswith('xyz') == True: return True
  for i in range(len(str)-3):
    if str[i:i+1] != '.'  and str[i+1:i+4].endswith('xyz') == True: return True
  return False

print(xyz_there('xyz.abc'))
print(xyz_there('xyz'))
print(xyz_there(''))
print(xyz_there('abc.xyzxyz'))
print(xyz_there('1.xyz.xyz2.xyz'))

File: java/codingBat/String-2/xyzThere.java
//Return true if the given string contains an appearance of "xyz" where the xyz is not directly preceeded by a period (.). So "xxyz" counts but "x.xyz" does not.
//xyzThere("abcxyz") → true
//xyzThere("abc.xyz") → false
//xyzThere("xyz.abc") → true

public class xyzThere{
    public static void main(String... args) {
        System.out.println(xyzThere("abcxyz"));
        System.out.println(xyzThere("abc.xyz"));
        System.out.println(xyzThere("xyz.abc"));
        System.out.println(xyzThere(".xyz.abc"));
        System.out.println(xyzThere(".axyz.abc"));
        System.out.println(xyzThere("abc.xyzxyz"));
    }
    public static boolean xyzThere(String str) {
        Boolean check = false;
        for (int x = 0; x < str.length() - 2; x ++) {
            if (x == 0) {
                if (str.substring(x, x + 3).equals("xyz")) {
                    check = true;
                }
            } else if (str.substring(x, x + 3).equals("xyz") && str.substring(x - 1, x).equals(".") == false) {
                    check = true;                
                }
            }
        return check;
    }
}